Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

S = sqrt(D/n_1  +  D/n_2)
D = 1/2(D1+D2)
L = x1 - x2

In [20]:
import numpy as np
import scipy.stats as stats

x = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
L = x.mean() - y.mean()
L 

-1.9000000000000057

In [21]:
M = x.var(ddof=1)
N = y.var(ddof=1)
D = (M + N) / 2
D

75.58333333333334

In [22]:
S = np.sqrt(D / len(x) + D / len(y))

S

3.8880157750022915

In [23]:
t = stats.t.ppf(0.975, (len(x) - 1) * 2)
L - t * S, L + t * S

(-10.068418034506857, 6.268418034506846)

Доверительный интервал для разности среднего роста родителей и детей 
{-10.068;  6.268}